src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportDestinationClient/InMemoryExportDestinationClient.cs

```diff
@@ -0,0 +1,81 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient
+{
+    public class InMemoryExportDestinationClient : IExportDestinationClient
+    {
+        private Dictionary<Uri, StringBuilder> _exportedData = new Dictionary<Uri, StringBuilder>();
+        private Dictionary<(Uri FileUri, uint PartId), Stream> _streamMappings = new Dictionary<(Uri FileUri, uint PartId), Stream>();
+
+        private readonly Uri _baseUri = new Uri("https://localhost:44348/");
+
+        public string DestinationType => "in-memory";
```

Comment:
The import spec (https://docs.google.com/document/d/1e9YYtDxc2Yed2RWU27jPLhfyIH0d6IH5Y9ligJUBFZE/edit#heading=h.bbcu6h9umbza) defines the destination type as "azure-blob" in all lower letters separated by hyphen so I followed similar patterns here as well. Even though this is extension we are building on top of the official bulk export spec, it would be nice to align with the upcoming standard.